src/test/java/Page_Objects/SongLanguagesPage.java

```diff
@@ -20,4 +23,38 @@ public SongLanguagesPage(WebDriver driver) {
     public void continueAndGoToSongList() {
         continueButton.click();
     }
+
+    public WebElement getLanguageElement(String language) {
+        return driver.findElement(By.cssSelector("[data-test=lang-" + language + "]"));
+    }
+
+    public WebElement getSongLanguageCheckbox(String language) {
+        return getLanguageElement(language).findElement(By.cssSelector("svg"));
+    }
+
+    public void ensureSongLanguageIsSelected(String language) {
+        if (isLanguageSelected(language)) {
+            getSongLanguageCheckbox(language).click();
+        }
+    }
+
+    public boolean isLanguageSelected(String language) {
+        WebElement languageWE = getSongLanguageCheckbox(language);
+        return isLanguageSelected(languageWE);
+    }
+
+    public boolean isLanguageSelected(WebElement language) {
+        return language.getAttribute("data-testid").equals("CheckBoxOutlineBlankIcon");

```

Comment:
This `CheckBoxOutlineBlankIcon` is actually present when the language is NOT selected (contrary to the name of the method)

Reply:
Good point, thanks. Done :)